### PR TITLE
Fix "collection.createIndex"

### PR DESCRIPTION
**Pull request by slonofanya**
Fix "collection.createIndex" fail, when "relationshipAttributeNames" isn't defined

### CODE REVIEW

**Comment by slonofanya** [comment]
@pmcnr-hx, @theninj4. Please look

**Comment by theninj4** [comment]
:+1: nice one! I wasn't aware it'd complain when called with an empty object, thanks for sharing :)

**Comment by pmcnr-hx** [comment]
Indeed. Well spotted @slonofanya! I do propose we simply return early if `relationshipAttributeNames` as a slightly more efficient and compact solution, though.

**Comment by pmcnr-hx** [comment]
What do you think of the following alternative @slonofanya:
```javascript
MongoStore.prototype._createIndexesForRelationships = function(collection, relationshipAttributeNames) {
  if (!relationshipAttributeNames || !relationshipAttributeNames.length) return;
  var index = relationshipAttributeNames.reduce(function(partialIndex, name) {
    partialIndex[name + ".id"] = 1;
    return partialIndex;
  }, {});
  collection.ensureIndex(index);
};
```
It's a bit more defensive by, in addition to checking for the empty array, also checking for `null` / `undefined` values for `relationshipAttributeNames` and returning immediately if that is the case. Also `.ensureIndex()` is probably a better alternative to `.createIndex()` to eliminating the possibility of duplicated indexes.
If the above works for you and you want to change your PR accordingly, for my part I would be happy to merge it as your contribution.

File: lib/mongoHandler.js [hunk 1]
```diff
@@ -147,15 +147,19 @@ MongoStore._unknownError = function(err) {
   };
 };
 
-
 MongoStore.prototype._createIndexesForRelationships = function(collection, relationshipAttributeNames) {
+  if (!(relationshipAttributeNames instanceof Array) || !relationshipAttributeNames.length) { return }
```
Comment (slonofanya) [inline lib/mongoHandler.js]: Add `;` after `return`

**Comment by slonofanya** [comment]
@pmcnr-hx, I updated the PR. I missed semicolon and now i cant to fix it. Please have a look

**Pull request merged**